### ISSUE TITLE
Why Bonsai and ProtoNN does not return logits but only the argmax of predictions?

### ISSUE THREAD

**Issue by swapnilsayansaha** [microsoft/EdgeML#256]
Normally, for any Tensorflow Model:

```
output = model(input)
```
returns the logits. However, Bonsai and ProtoNN is returning the argmax of the predictions after reconstructing the model from pretrained Bonsai. Example code:

```
Z = np.load( glob.glob(model_dir + "/**/Z.npy", recursive = True)[0], allow_pickle=True )
W = np.load( glob.glob(model_dir + "/**/W.npy", recursive = True)[0], allow_pickle=True )
V = np.load( glob.glob(model_dir + "/**/V.npy", recursive = True)[0], allow_pickle=True )
T = np.load( glob.glob(model_dir + "/**/T.npy", recursive = True)[0], allow_pickle=True )
hyperparams = np.load(glob.glob(model_dir + "/**/hyperParam.npy", 
                            recursive = True)[0], allow_pickle=True ).item()

n_dim = hyperparams['dataDim']
projectionDimension = hyperparams['projDim']
numClasses = hyperparams['numClasses']
depth = hyperparams['depth']
sigma = hyperparams['sigma']

dense = BonsaiLayer( numClasses, n_dim, projectionDimension, depth, sigma )
model = keras.Sequential([
keras.layers.InputLayer(n_dim),
dense
])

dummy_tensor = tf.convert_to_tensor( np.zeros((1,n_dim), np.float32) )
out_tensor = model(dummy_tensor)
model.summary()
dense.set_weights( [Z, W, V, T] )
model.compile()
```
```
audio = Xtest[0,:].reshape(1,len(Xtest[0,:])) #a 1X10 matrix
output = model(audio) #this returns argmax of logits and NOT logits.

```

This prevents ```tape.gradient()``` to be useable.

**Comment by swapnilsayansaha** [comment]
Never mind I modified the bonsai layer to return logits.
